server/protocols/jmap-draft/src/test/java/org/apache/james/jmap/draft/model/message/view/MessageFullViewFactoryTest.java

```diff
@@ -157,6 +157,20 @@ void fromMessageResultsShouldStorePreviewWhenPresent() throws Exception {
             .isEqualTo(Preview.from("blabla bloblo"));
     }
 
+    @Test
+    void fromMessageResultsShouldUsePreviewWhenAlreadyComputed() throws Exception {
+        String preview = "my pre computed preview";
+        Mono.from(previewStore.store(message1.getMessageId(), Preview.from(preview)))
+            .block();
+
+        List<MessageResult> messages = messageIdManager
+            .getMessages(ImmutableList.of(message1.getMessageId()), FetchGroup.FULL_CONTENT, session);
+
+        MessageFullView actual = messageFullViewFactory.fromMessageResults(messages);
+        assertThat(actual.getPreview()).isEqualTo(PreviewDTO.of(preview));
+
+    }
+
```

Comment:
We are missing this test:

```
Given a message stored in the mailbox-api without a preview
When messageFullViewFactory loads message results
Then the preview is stored (the projection is updated)
```

Reply:
isn't it `fromMessageResultsShouldStorePreviewWhenPresent` ?